src/timings/TimingsHandler.php

```diff
@@ -332,4 +342,53 @@ public function reset() : void{
 		$this->recordsByParent = [];
 		$this->timingDepth = 0;
 	}
+
+	/**
+	 * Creates a timings report file locally in the provided file path.
+	 * Collects timings data and returns a promise that resolves with the created file.
+	 *
+	 * @param string      $filePath directory path to the timings folder.
+	 * @param string|null $fileName Optional custom file name, If null, uses default timings file naming
+	 *
+	 * @phpstan-return Promise<string>
+	 */
+	public static function createReportFile(string $filePath, ?string $fileName = null) : Promise{
+		$timingsPromise = self::requestPrintTimings();
+
+		/** @var PromiseResolver<string> $resolver */
+		$resolver = new PromiseResolver();
+
+		$timingsPromise->onCompletion(
+			function(array $lines) use ($fileName, $filePath, $resolver) : void{
+				$date = date('Y-m-d_H.i.s_T');
+				if($fileName === null){
+					$timingsName = "timings_{$date}";
+				}else{
+					$timingsName = "{$fileName}";
+				}
+				if(!file_exists($filePath)){
+					mkdir($filePath, 0777, true);
+				}
+				$timingsFile = Path::join($filePath, $timingsName . ".txt");
+				if(file_exists($timingsFile)){
+					if(!is_file($timingsFile)){
```

Comment:
Not sure what the point of this branch is. It's rejecting the promise either way.